server/aws-lsp-codewhisperer/src/language-server/agenticChat/agenticChatController.ts

```diff
@@ -448,6 +449,11 @@ export class AgenticChatController implements ChatHandlers {
                 ? []
                 : this.#chatHistoryDb.getMessages(tabId)
 
+            // Phase 3: Request Execution
```

Comment:
what was the rational about moving these before the current user message was added to history?

Reply:
We were thinking that it's possible for the request to fail which means we shouldn't add it to history. So I think it's better to add it after the request is complete